substrate/frame/collective/src/lib.rs

```diff
@@ -173,6 +173,138 @@ pub struct Votes<AccountId, BlockNumber> {
 	end: BlockNumber,
 }
 
+/// Types implementing various cost strategies for a given proposal count.
+///
+/// These types implement [Convert](sp_runtime::traits::Convert) trait and can be used with types
+/// like [HoldConsideration](`frame_support::traits::fungible::HoldConsideration`) implementing
+/// [Consideration](`frame_support::traits::Consideration`) trait.
+///
+/// ### Example:
+///
+/// 1. Linear increasing with helper types.
+/// #[doc = docify::embed!("src/tests.rs", deposit_types_with_linear_work)]
+///
+/// 2. Geometrically increasing with helper types.
+/// #[doc = docify::embed!("src/tests.rs", deposit_types_with_geometric_work)]
+///
+/// 3. Geometrically increasing with rounding.
+/// #[doc = docify::embed!("src/tests.rs", deposit_round_with_geometric_work)]
+pub mod deposit {
+	use core::marker::PhantomData;
+	use sp_core::Get;
+	use sp_runtime::{traits::Convert, FixedPointNumber, FixedU128, Saturating};
+
+	/// Constant deposit amount regardless of current proposal count.
+	/// Returns `None` if configured with zero deposit.
+	pub struct Constant<Deposit>(PhantomData<Deposit>);
+	impl<Deposit, Balance> Convert<u32, Balance> for Constant<Deposit>
+	where
+		Deposit: Get<Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(_: u32) -> Balance {
+			Deposit::get()
+		}
+	}
+
+	/// Linear increasing with some offset.
+	/// f(x) = ax + b, a = `Slope`, x = `proposal_count`, b = `Offset`.
+	pub struct Linear<Slope, Offset>(PhantomData<(Slope, Offset)>);
```

Comment:
Idea: You could add an `pub struct Offset<` that can then be combined with the other ones instead of hard-coding the offset here.

Reply:
If fine by you I keep it as is for now. If we need offset later, we can add it.